integration_tests/utils.py

```diff
@@ -383,61 +388,143 @@ def __init__(self, private_key=KEYS["validator"], chain_id=777):
 
         self.contract = None
         self._contract_hash = None
+        self.w3 = None
 
     def deploy(self, w3):
         "Deploy Greeter contract on `w3` and return the transaction hash."
         if self.contract is None:
-            contract = w3.eth.contract(abi=self.abi, bytecode=self.bytecode)
-            self.nonce = w3.eth.get_transaction_count(self.address)
+            self.w3 = w3
+            contract = self.w3.eth.contract(abi=self.abi, bytecode=self.bytecode)
+            self.nonce = self.w3.eth.get_transaction_count(self.address)
 
             transaction = contract.constructor().buildTransaction(
                 {"chainId": self.chain_id, "from": self.address, "nonce": self.nonce}
             )
 
-            signed_txn = w3.eth.account.sign_transaction(
-                transaction, private_key=self.private_key
-            )
-
-            # print("Deploying contract...")
-            # Send this signed transaction
-            self._contract_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)
+            tx_hash, tx_receipt = self.sign(transaction).send().wait()
```

Comment:
I think we can use `send_transaction(tx)` here?

Reply:
let me replace with send_transaction.